pkg/resources/oauth_integration_for_custom_clients.go

```diff
@@ -0,0 +1,666 @@
+			return diag.FromErr(err)
+		}
+
+		networkPolicy, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "NETWORK_POLICY"
+		})
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to find network policy, err = %w", err))
+		}
+		if err := d.Set("network_policy", sdk.NewAccountObjectIdentifier(networkPolicy.Value).Name()); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if withExternalChangesMarking {
+			if err = handleExternalChangesToObjectInShow(d,
+				showMapping{"enabled", "enabled", integration.Enabled, strconv.FormatBool(integration.Enabled), nil},
+			); err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthAllowNonTlsRedirectUri, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_ALLOW_NON_TLS_REDIRECT_URI"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthEnforcePkce, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_ENFORCE_PKCE"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthUseSecondaryRoles, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_USE_SECONDARY_ROLES"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthIssueRefreshTokens, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_ISSUE_REFRESH_TOKENS"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthRefreshTokenValidity, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_REFRESH_TOKEN_VALIDITY"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			if err = handleExternalChangesToObjectInDescribe(d,
+				describeMapping{"oauth_allow_non_tls_redirect_uri", "oauth_allow_non_tls_redirect_uri", oauthAllowNonTlsRedirectUri.Value, oauthAllowNonTlsRedirectUri.Value, nil},
+				describeMapping{"oauth_enforce_pkce", "oauth_enforce_pkce", oauthEnforcePkce.Value, oauthEnforcePkce.Value, nil},
+				describeMapping{"oauth_use_secondary_roles", "oauth_use_secondary_roles", oauthUseSecondaryRoles.Value, oauthUseSecondaryRoles.Value, nil},
+				describeMapping{"oauth_issue_refresh_tokens", "oauth_issue_refresh_tokens", oauthIssueRefreshTokens.Value, oauthIssueRefreshTokens.Value, nil},
+				describeMapping{"oauth_refresh_token_validity", "oauth_refresh_token_validity", oauthRefreshTokenValidity.Value, oauthRefreshTokenValidity.Value, nil},
+			); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		if err = setStateToValuesFromConfig(d, oauthIntegrationForCustomClientsSchema, []string{
```

Comment:
are these all? what about e.g. network_policy?

Reply:
network policy is set above "the if", so I'm guessing it's obsolete here